### PR TITLE
Various Window Fixes

### LINKED ISSUE #620: Window Switcher lets windows grow indefinetly
https://user-images.githubusercontent.com/14252419/211862978-c982ae1c-7838-49c9-b369-6cdbbdcd64d9.mp4

Additionally, windows never get unblurred.

### PR DESCRIPTION
**Pull request by linusha**
Closes #620.

fixed_window_switcher

Also fixes a problem where windows would change their position/bounds when being minimized/uncollapsed.
fixed_minimization

Resizing windows with the left and top resizers will no longer result in changed positions or bounds.
fixed_resizing

Maximizing windows will now behave smarter. When a window is maximized (double click on the window title) and is currently not displayed in fullscreen, it will go into fullscreen. Double clicking a window that is displayed in full screen will change its position and extent back to the last values **before** the window was displayed in fullscreen.
improved_maximization

### CODE REVIEW

File: lively.components/window.js [hunk 1]
```diff
@@ -489,52 +473,84 @@ export default class Window extends Morph {
...
   }
 
-  toggleMaximize () { if (!this.minimized) this.maximized = !this.maximized; }
+  beginResizing (evt) {
+    this.startPos = evt.position;
+    this.startBounds = this.position.extent(this.extent);
+    this.resizingTfm = this.getGlobalTransform().inverse();
+  }
+
+  toggleMaximize () { if (!this.minimized) this.applyMaximize(); }
```
Comment (merryman) [inline lively.components/window.js:528]: Does not really toggle right? Just maximizes and that's it.

File: lively.graphics/geometry-2d.js [hunk 2]
```diff
@@ -771,17 +783,24 @@ export class Rectangle {
...
     }
 
     return nearest;
   }
 
+  withWholeNumbers () {
+    const { x, y } = new Point(this.x, this.y).roundTo(1);
+    const { x: width, y: height } = new Point(this.width, this.height).roundTo(1);
+    return new Rectangle(x, y, width, height);
+  }
```
Comment (merryman) [inline lively.graphics/geometry-2d.js:802]: May I suggest replacing this new method with a *higher level* `roundTo(n)` for rectangles?
That way we could cover more than just rounding to whole number via this method.

```suggestion
  roundTo (n) {
    const { x, y } = new Point(this.x, this.y).roundTo(n);
    const { x: width, y: height } = new Point(this.width, this.height).roundTo(n);
    return new Rectangle(x, y, width, height);
  }
```

**Review by merryman — CHANGES_REQUESTED** [review]

File: lively.graphics/geometry-2d.js (hunk 2 above)
Comment (linusha) [reply lively.graphics/geometry-2d.js:802]: Done :heavy_check_mark:

File: lively.components/window.js [hunk 3]
```diff
@@ -89,22 +90,25 @@ export default class Window extends Morph {
...
-        set (isMaximized) {
-          this.setProperty('maximized', isMaximized);
-          this.applyMaximize();
+          if (changed) this.applyMinimize();
         }
       }
     };
   }
 
+  static maximizedBounds () {
```
[the remark this answers is not in the document]
Comment (linusha) [reply lively.components/window.js:104]: Done :heavy_check_mark:

**Review by merryman — APPROVED** [review]

**Pull request merged**